File: eRxWeb/ServerModel/HomeAddressModel.cs

using Allscripts.ePrescribe.Data;
using System.Collections.Generic;

namespace eRxWeb.ServerModel
{
    public class HomeAddressStartUpModel
    {
        public List<StateModel> States { get; set; }
    }

    public class HomeAddressSaveRequest
    {
        public string HomeAddress { get; set; }
        public string HomeAddress2 { get; set; }
        public string City { get; set; }
        public string State { get; set; }
        public string ZipCode { get; set; }
    }

    public class HomeAddressSaveModel
    {
        public bool IsSaveSuccessful { get; set; }
        public string RedirectUrl { get; set; }
    }
}

File: eRxWeb/SPA/Model/home-address.model.ts
export class HomeAddressStartUpModel {
    States: State[];
}
export class State {
    State: string;
    Description: string
}

export class HomeAddressDetail {
    HomeAddress: string;
    HomeAddress2: string;
    City: string;
    State: string;
    ZipCode: string;
}

export class HomeAddressResponse {
    IsSaveSuccessful: boolean
    RedirectUrl: string;
}